src/main/java/de/tum/in/www1/artemis/service/connectors/localci/buildagent/LocalCIDockerService.java

```diff
@@ -285,15 +294,102 @@ private void checkImageArchitecture(String imageName, InspectImageResponse inspe
 
     @Scheduled(cron = "${artemis.continuous-integration.image-cleanup.cleanup-schedule-time:0 0 3 * * *}")
     public void deleteOldDockerImages() {
-
         if (!imageCleanupEnabled) {
             log.info("Docker image cleanup is disabled");
             return;
         }
 
+        Set<String> imageNames = getUnusedDockerImages();
+
         // Get map of docker images and their last build dates
         IMap<String, ZonedDateTime> dockerImageCleanupInfo = hazelcastInstance.getMap("dockerImageCleanupInfo");
 
+        // Delete images that have not been used for more than imageExpiryDays days
+        for (String dockerImage : dockerImageCleanupInfo.keySet()) {
+            if (imageNames.contains(dockerImage)) {
+                if (dockerImageCleanupInfo.get(dockerImage).isBefore(ZonedDateTime.now().minusDays(imageExpiryDays))) {
+                    log.info("Deleting docker image {}", dockerImage);
+                    try {
+                        dockerClient.removeImageCmd(dockerImage).exec();
+                    }
+                    catch (NotFoundException e) {
+                        log.warn("Docker image {} not found during cleanup", dockerImage);
+                    }
+                }
+            }
+        }
+    }
+
+    /**
+     * Checks for available disk space and triggers the cleanup of old Docker images if the available space falls below
+     * {@link LocalCIDockerService#imageCleanupDiskSpaceThresholdMb}.
+     *
+     * @implNote - We use the Docker root directory to check disk space availability. This is in case the Docker images are stored on a separate partition.
+     *           - We need to iterate over the map entries since don't remove the oldest image from the map.
+     */
+
+    @Scheduled(fixedRateString = "${artemis.continuous-integration.image-cleanup.disk-space-check-interval-ms:3600000}", initialDelayString = "${artemis.continuous-integration.image-cleanup.disk-space-check-interval-ms:3600000}")
```

Comment:
A considerable number like `3600000` is complex to read/understand and requires further developers or reviewers to calculate them repeatedly to realize that it´s just `1h`. Is there an option to use hours or minutes in the configuration and perform a respective calculation for configuring the `fixedRate`?